app/code/community/Hps/Securesubmit/Model/Payment.php

```diff
@@ -252,6 +243,29 @@ private function _authorize(Varien_Object $payment, $amount, $capture)
         return $this;
     }
 
+    /**
+     * @param Varien_Object|Mage_Sales_Model_Order_Payment  $payment
+     * @param float                                         $amount
+     * @param HpsReportTransactionDetails|HpsAuthorization  $response
+     * @param Mage_Payment_Model_Method_Abstract::STATUS_UNKNOWN|STATUS_APPROVED|STATUS_ERROR|STATUS_DECLINED|STATUS_VOID|STATUS_SUCCESS                                   $status
+     */
+    protected function closeTransaction($payment, $amount, $response, $status = self::STATUS_APPROVED){
+        $payment->setStatus($status);
+        $payment->setAmount($amount);
+        if (property_exists($response,'cardType')){
+            $payment->setCcType($response->cardType);
```

Comment:
@ttnnkkrr I'd prefer this be set with the card type obtained from the frontend (see saving multi-use tokens in this file) if possible to cover when the gateway returns the network used instead of the card's actual type.

Also, does this need to be formatted in any specific way for Magento? Is it displayed for the admin anywhere?